### ISSUE TITLE
feature request: support `override-dependencies` for inline scripts

### ISSUE THREAD

**Issue by Fogapod** [astral-sh/uv#9232]
uv: `0.5.2`
According to docs inline scripts support `[tool.uv]` settings but override-dependencies doesn't work for me:
```
error: TOML parse error at line 7, column 1
  |
7 | [tool.uv]
  | ^^^^^^^^^
unknown field `override-dependencies`
```

reproducer:
```python
# /// script
# requires-python = ">=3.12"
# dependencies = [
#     "pymilvus",
# ]
#
# [tool.uv]
# override-dependencies = [
#     "milvus-lite; sys_platform == 'never'",
# ]
# ///
```

**Comment by charliermarsh** [comment]
This is fixed in v0.5.3, which hasn't gone out yet: https://github.com/astral-sh/uv/pull/9229

**Comment by charliermarsh** [comment]
Relevant PR is: https://github.com/astral-sh/uv/pull/9162
